packages/app/src/systems/DApp/hooks/useSignatureRequest.tsx

```diff
@@ -0,0 +1,49 @@
+import { useMachine, useSelector } from '@xstate/react';
+
+import type { SignMachineState, UnlockMachineState } from '../machines';
+import { signMachine } from '../machines';
+
+import { useAccount } from '~/systems/Account';
+
+const selectors = {
+  isUnlocking: (state: SignMachineState) => state.matches('unlocking'),
+  signedMessage: (state: SignMachineState) => state.context.signedMessage,
+  isUnlockingLoading: (state: UnlockMachineState) => state.matches('unlocking'),
+};
+
+export function useSignatureRequest() {
+  const { account } = useAccount();
+  const [state, send, service] = useMachine(signMachine);
```

Comment:
Why `useMachine` instead of `useInterpreter`?

Reply:
for local machines we normally use `useMachine`
it's good because it returns `[state, send, service]`
`useInterpret` returns only `service`


we need access to child `state.children.unlock` machine here

we used before `useInterpret` for global machines only, but was replaced by new `store` implementation


although `useInterpret` seems better for performance:
> The useInterpret is useful when you want fine-grained control, e.g. minimize re-renders. In contrast to useMachine that would flush each update from the machine to the React component, useInterpret instead returns a static reference (to just the interpreted machine) which will not rerender when its state changes.


if we find other way of accessing child machines, could try moving to `useInterpret`

Reply:
Does the state returned from the useSelector(service) doesn't have access to the children's state?

Reply:
it has access but doesn't update as expected.

needs to pass directly the child machine to `useSelector` in order to update